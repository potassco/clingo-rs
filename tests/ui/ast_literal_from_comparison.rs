// ast_literal_from_comparison
use clingo::*;

fn main() {
    let loc = ast::Location::default();
    let sym = Symbol::create_id("test", true).unwrap();
    let term1: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
    let term2: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
    let guard = ast::guard(ast::ComparisonOperator::LessEqual, term1).unwrap();
    let guards = [guard];
    let comp = ast::comparison(term2, &guards).unwrap();
    let lit = ast::basic_literal_from_comparison(&loc, ast::Sign::NoSign, comp).unwrap();

    let _end = (lit, comp);
}
