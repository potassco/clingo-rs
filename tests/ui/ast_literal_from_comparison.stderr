error[E0382]: use of moved value: `comp`
  --> tests/ui/ast_literal_from_comparison.rs:14:22
   |
11 |     let comp = ast::comparison(term2, &guards).unwrap();
   |         ---- move occurs because `comp` has type `Comparison<'_>`, which does not implement the `Copy` trait
12 |     let lit = ast::basic_literal_from_comparison(&loc, ast::Sign::NoSign, comp).unwrap();
   |                                                                           ---- value moved here
13 |
14 |     let _end = (lit, comp);
   |                      ^^^^ value used here after move
   |
help: consider cloning the value if the performance cost is acceptable
   |
12 |     let lit = ast::basic_literal_from_comparison(&loc, ast::Sign::NoSign, comp.clone()).unwrap();
   |                                                                               ++++++++
